packages/react-core/src/components/LoginPage/LoginForm.tsx

```diff
@@ -67,59 +79,84 @@ export const LoginForm: React.FunctionComponent<LoginFormProps> = ({
-      fieldId="pf-login-username-id"
-    >
-      <TextInput
-        autoFocus={!noAutoFocus}
-        id="pf-login-username-id"
+}: LoginFormProps) => {
+  const [passwordHidden, setPasswordHidden] = React.useState(true);
+  return (
+    <Form className={className} {...props}>
+      <FormHelperText isError={!isValidUsername || !isValidPassword} isHidden={!showHelperText} icon={helperTextIcon}>
+        {helperText}
+      </FormHelperText>
+      <FormGroup
+        label={usernameLabel}
         isRequired
         validated={isValidUsername ? ValidatedOptions.default : ValidatedOptions.error}
-        type="text"
-        name="pf-login-username-id"
-        value={usernameValue}
-        onChange={onChangeUsername}
-      />
-    </FormGroup>
-    <FormGroup
-      label={passwordLabel}
-      isRequired
-      validated={isValidPassword ? ValidatedOptions.default : ValidatedOptions.error}
-      fieldId="pf-login-password-id"
-    >
-      <TextInput
+        fieldId="pf-login-username-id"
+      >
+        <TextInput
+          autoFocus={!noAutoFocus}
+          id="pf-login-username-id"
+          isRequired
+          validated={isValidUsername ? ValidatedOptions.default : ValidatedOptions.error}
+          type="text"
+          name="pf-login-username-id"
+          value={usernameValue}
+          onChange={onChangeUsername}
+        />
+      </FormGroup>
+      <FormGroup
+        label={passwordLabel}
         isRequired
-        type="password"
-        id="pf-login-password-id"
-        name="pf-login-password-id"
         validated={isValidPassword ? ValidatedOptions.default : ValidatedOptions.error}
-        value={passwordValue}
-        onChange={onChangePassword}
-      />
-    </FormGroup>
-    {rememberMeLabel.length > 0 && (
-      <FormGroup fieldId="pf-login-remember-me-id">
-        <Checkbox
-          id="pf-login-remember-me-id"
-          label={rememberMeLabel}
-          isChecked={isRememberMeChecked}
-          onChange={onChangeRememberMe}
-        />
+        fieldId="pf-login-password-id"
+      >
+        {isShowPasswordEnabled && (
+          <InputGroup>
+            <TextInput
+              isRequired
+              type={passwordHidden ? 'password' : 'text'}
+              id="pf-login-password-id"
+              name="pf-login-password-id"
+              validated={isValidPassword ? ValidatedOptions.default : ValidatedOptions.error}
+              value={passwordValue}
+              onChange={onChangePassword}
+            />
+            <Button
+              variant="control"
+              onClick={() => setPasswordHidden(!passwordHidden)}
+              aria-label={passwordHidden ? showPasswordLabel : hidePasswordLabel}
+            >
+              {passwordHidden ? <EyeIcon /> : <EyeSlashIcon />}
+            </Button>
+          </InputGroup>
+        )}
+        {!isShowPasswordEnabled && (
+          <TextInput
```

Comment:
this is pretty much duplicate of lines 114-122.  can you make cont of that code and use that instead?